### ISSUE TITLE
False positive on S1481 with string interpolation

### ISSUE THREAD

**Issue by RowlandShaw** [SonarSource/sonar-dotnet#2402]
### Description

False positive on S1481 where the only use of the variable is in a string interpolation

### Repro steps

```
Public Sub FalsePositive
        Dim falsePositive As String = "Y"
        Console.WriteLine("{falsePositive}")
End Sub
```
In this code, `falsePositive` variable is deemed as unused, even though it clearly is

### Expected behavior

`falsePositive` should not be identified as unused

### Actual behavior

Warning	S1481	Remove the unused local variable 'falsePositive'.	MyProject	Myfile.vb	209	Active

### Known workarounds

Change to use longhand String.Format syntax

### Related information

* SonarLint Version 4.9.0.4316
* Visual Studio Version 2017/2019

**Comment by RowlandShaw** [comment]
Important

You cannot have any white space between the $ and the " that starts the string. Doing so causes a compiler error.

(user error)
